bundles/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/config/IntegraStateBindingConfig.java

```diff
@@ -0,0 +1,258 @@
+import org.openhab.core.types.State;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+/**
+ * This class implements binding configuration for all items that represents
+ * Integra zones/inputs/outputs state.
+ * 
+ * @author Krzysztof Goworek
+ * @since 1.7.0
+ */
+public class IntegraStateBindingConfig implements SatelBindingConfig {
+
+	private static final DecimalType DECIMAL_ONE = new DecimalType(1);
+
+	private StateType stateType;
+	private int objectNumber;
+	private Map<String, String> options;
+
+	private IntegraStateBindingConfig(StateType stateType, int objectNumber, Map<String, String> options) {
+		this.stateType = stateType;
+		this.objectNumber = objectNumber;
+		this.options = options;
+	}
+
+	/**
+	 * Parses given binding configuration and creates configuration object.
+	 * 
+	 * @param bindingConfig
+	 *            config to parse
+	 * @return parsed config object or <code>null</code> if config does not
+	 *         match
+	 * @throws BindingConfigParseException
+	 *             in case of parse errors
+	 */
+	public static IntegraStateBindingConfig parseConfig(String bindingConfig) throws BindingConfigParseException {
+		String[] configElements = bindingConfig.split(":");
+		int idx = 0;
+		ObjectType objectType;
+
+		// parse object type, mandatory
+		try {
+			objectType = ObjectType.valueOf(configElements[idx++]);
+		} catch (Exception e) {
+			// wrong config type, skip parsing
+			return null;
+		}
+
+		// parse state type, mandatory except for output
+		StateType stateType = null;
+		int objectNumber = 0;
+
+		try {
+			switch (objectType) {
+			case input:
+				stateType = InputState.valueOf(configElements[idx++]);
+				break;
+			case zone:
+				stateType = ZoneState.valueOf(configElements[idx++]);
+				break;
+			case output:
+				stateType = OutputState.output;
+				break;
+			case doors:
+				stateType = DoorsState.valueOf(configElements[idx++]);
+				break;
+			}
+		} catch (Exception e) {
+			throw new BindingConfigParseException(String.format("Invalid state type: {}", bindingConfig));
+		}
+
+		// parse object number, if provided
+		if (idx < configElements.length) {
+			try {
+				objectNumber = Integer.parseInt(configElements[idx++]);
+			} catch (NumberFormatException e) {
+				throw new BindingConfigParseException(String.format("Invalid object number: {}", bindingConfig));
+			}
+		}
+
+		// parse options: comma separated pairs of <name>=<value>
+		Map<String, String> options = new HashMap<String, String>();
+		if (idx < configElements.length) {
+			for (String option : configElements[idx++].split(",")) {
+				if (option.contains("=")) {
+					String[] keyVal = option.split("=", 2);
+					options.put(keyVal[0], keyVal[1]);
+				} else {
+					options.put(option, "");
+				}
+			}
+		}
+
+		if (idx < configElements.length) {
+			// if anything left, throw exception
+			throw new BindingConfigParseException(String.format("Too many elements: {}", bindingConfig));
```

Comment:
since you obviously know the expected number of elements please enhance the Message to guide the user "Too many elements in BindingConfig {}. Expected x but found y"